src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -1175,6 +1206,26 @@ public void onMessage(SignalServiceEnvelope envelope) {
         }
     }
 
+    private boolean isMessageBlocked(SignalServiceEnvelope envelope, SignalServiceContent content) {
+        SignalServiceAddress source = envelope.getSourceAddress();
```

Comment:
With signal sealed messages the envelope doesn't always contain the actual source address.
This needs an additional check:
```java
            SignalServiceAddress source;
            if (!envelope.isUnidentifiedSender() && envelope.hasSource()) {
                source = envelope.getSourceAddress();
            } else if (content != null) {
                source = content.getSender();
            }